

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Command-line API &mdash; iMARGI Pipeline 1.1 documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Guides for Issues on Windows and macOS System" href="faq_win_mac.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> iMARGI Pipeline
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="index.html">Overview</a></li>
</ul>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Tools and Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quick_example.html">Quick Start Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="step_by_step_illustration.html">Step-by-step Illustration</a></li>
<li class="toctree-l1"><a class="reference internal" href="further_analysis.html">Further Analysis and Visualization Guides</a></li>
<li class="toctree-l1"><a class="reference internal" href="technical_note.html">Technical Notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq_win_mac.html">Guides for Issues on Windows and macOS System</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Command-line API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#imargi-wrapper-sh">imargi_wrapper.sh</a></li>
<li class="toctree-l2"><a class="reference internal" href="#imargi-clean-sh">imargi_clean.sh</a></li>
<li class="toctree-l2"><a class="reference internal" href="#imargi-rsfrags-sh">imargi_rsfrags.sh</a></li>
<li class="toctree-l2"><a class="reference internal" href="#imargi-parse-sh">imargi_parse.sh</a></li>
<li class="toctree-l2"><a class="reference internal" href="#imargi-stats-sh">imargi_stats.sh</a></li>
<li class="toctree-l2"><a class="reference internal" href="#imargi-distfilter-sh">imargi_distfilter.sh</a></li>
<li class="toctree-l2"><a class="reference internal" href="#imargi-convert-sh">imargi_convert.sh</a></li>
<li class="toctree-l2"><a class="reference internal" href="#imargi-annotate-sh">imargi_annotate.sh</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">iMARGI Pipeline</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Command-line API</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="command-line-api">
<h1>Command-line API<a class="headerlink" href="#command-line-api" title="Permalink to this headline">¶</a></h1>
<p>We created several script tools. Here we show the usage and source code of all these tools.</p>
<ul class="simple">
<li><a class="reference external" href="#command-line-api">Command-line API</a><ul>
<li><a class="reference external" href="#imargi_wrappersh">imargi_wrapper.sh</a></li>
<li><a class="reference external" href="#imargi_cleansh">imargi_clean.sh</a></li>
<li><a class="reference external" href="#imargi_rsfragssh">imargi_rsfrags.sh</a></li>
<li><a class="reference external" href="#imargi_parsesh">imargi_parse.sh</a></li>
<li><a class="reference external" href="#imargi_statssh">imargi_stats.sh</a></li>
<li><a class="reference external" href="#imargi_distfiltersh">imargi_distfilter.sh</a></li>
<li><a class="reference external" href="#imargi_convertsh">imargi_convert.sh</a></li>
<li><a class="reference external" href="#imargi_annotatesh">imargi_annotate.sh</a></li>
</ul>
</li>
</ul>
<div class="section" id="imargi-wrapper-sh">
<h2>imargi_wrapper.sh<a class="headerlink" href="#imargi-wrapper-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_wrapper.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: $PROGNAME [-r &lt;ref_name&gt;] [-N &lt;base_name&gt;] [-g &lt;ref_fasta&gt;]
                     [-c &lt;chromSize_file&gt;] [-i &lt;bwa_index&gt;] [-R &lt;restrict_sites&gt;] 
                     [-Q &lt;min_mapq&gt;] [-G &lt;max_inter_align_gap&gt;]
                     [-O &lt;offset_restriction_site&gt;] [-M &lt;max_ligation_size&gt;] [-t &lt;threads&gt;]
                     [-1 &lt;fastq.gz_R1&gt;] [-2 &lt;fastq.gz_R2&gt;]
                     [-o &lt;output_dir&gt;]
    
    Dependency: seqtk, samtools, bwa, pairtools, pbgzip

    This is an all-in-one wrapper script of iMARGI Pipeline.

    -r : Reference assembly name, such as &quot;hg38&quot;
    -N : Base name for ouput result. Such as -N HEK_iMARGI, then the output file will be renamed using the base name,
         such as final_HEK_iMARGI.paris.gz, HEK_iMARGI.bam.
    -g : Reference genome fasta file. It will be used to generate ref files of &#39;-c&#39;, &#39;-i&#39; and &#39;-R&#39;. If any of them are
         not supplied. It will be generated from the reference fasta file.
    -c : Chromosome size file.
    -i : bwa index
    -R : DNA restriction enzyme digestion fragments bed file.
    -Q : Min MAPQ value for parsing as unique mapping. Default 1.
    -G : Max inter align gap for pairtools parsing. Default 20. It will allow R1 5&#39; end clipping.
    -O : Max offset bases for filtering pairs based on R2 5&#39; end positions to restriction sites. Default 3.
    -M : Max size of ligation fragment for sequencing. It&#39;s used for filtering unligated DNA sequence.
    -t : Max CPU threads for parallelized processing, at least 4. (Default 8)
    -1 : R1 fastq.gz file, if there are multiple files, just separated with space or use wildcard,
         such as &#39;-1 lane1_R1.fq.gz lane2_R1.fq.gz&#39;, or &#39;-1  lane*_R1.fq.gz&#39;.
    -2 : R2 fastq.gz file, if there are multiple files, just separated with space or use wildcard,
         such as &#39;-2 lane1_R2.fq.gz lane2_R2.fq.gz&#39;, or &#39;-2  lane*_R2.fq.gz&#39;.
    -o : Output directoy
    -h : Show usage help
</pre></div>
</div>
</div>
<div class="section" id="imargi-clean-sh">
<h2>imargi_clean.sh<a class="headerlink" href="#imargi-clean-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_clean.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: <span class="nv">$PROGNAME</span> <span class="o">[</span>-1 &lt;fastq.gz_R1&gt;<span class="o">]</span> <span class="o">[</span>-2 &lt;fastq.gz_R2&gt;<span class="o">]</span> <span class="o">[</span>-N &lt;base_name&gt;<span class="o">]</span> <span class="o">[</span>-o &lt;output_dir&gt;<span class="o">]</span> <span class="o">[</span>-t &lt;threads&gt;<span class="o">]</span>
    
    Dependencies: seqtk, gzip, zcat, awk, parallel

    This script will clean the paired reads <span class="o">(</span>R1 and R2<span class="o">)</span> of iMARGI sequencing Fastq data. According to the iMARGI design,
    RNA end reads <span class="o">(</span>R1<span class="o">)</span> start with <span class="m">2</span> random based. We need to remove the first <span class="m">2</span> bases of R1 <span class="k">for</span> better mapping. 
    If you provided multiple input files <span class="o">(</span>different lanes<span class="o">)</span> in <span class="s1">&#39;-1&#39;</span> and <span class="s1">&#39;-2&#39;</span> with <span class="s1">&#39;,&#39;</span> separator or contains wildcard,
    <span class="k">then</span> the output will merge multi-lanes fastq files to one clean fastq file.

    The input fastq files must be gzip files, i.e., fastq.gz or fq.gz. The output files are also gzipped files fastq.gz.

    -1 : R1 fastq.gz file, <span class="k">if</span> there are multiple files, just separated with space or use wildcard,
         such as <span class="s1">&#39;-1 lane1_R1.fq.gz lane2_R1.fq.gz&#39;</span>, or <span class="s1">&#39;-1  lane*_R1.fq.gz&#39;</span>.
    -2 : R2 fastq.gz file, <span class="k">if</span> there are multiple files, just separated with space or use wildcard,
         such as <span class="s1">&#39;-2 lane1_R2.fq.gz lane2_R2.fq.gz&#39;</span>, or <span class="s1">&#39;-2  lane*_R2.fq.gz&#39;</span>.
    -N : Base name <span class="k">for</span> ouput result. Such as -N HEK_iMARGI, <span class="k">then</span> output cleaned and merged fastq.gz file will be
         renamed using the base name.
    -o : Output directoy
    -t : Max CPU threads <span class="k">for</span> parallelized processing, at least <span class="m">4</span>. <span class="o">(</span>Default <span class="m">8</span><span class="o">)</span>
    -h : Show usage <span class="nb">help</span>
</pre></div>
</div>
</div>
<div class="section" id="imargi-rsfrags-sh">
<h2>imargi_rsfrags.sh<a class="headerlink" href="#imargi-rsfrags-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_rsfrags.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: $PROGNAME [-r &lt;ref_fasta&gt;] [-c &lt;chromSize_file&gt;] [-e &lt;enzyme_name&gt;] [-C &lt;cut_position&gt;] [-o &lt;output_dir&gt;] 

    Dependency: cooler
    This script use cooler digest to generate the restriction Enzyme digested fragments bed file for iMARGI
    -r : Reference genome fasta file
    -c : Chromosome size file.
    -e : Enzyme name, we use AluI in iMARGI.
    -C : Cut position right offset of the Enzyme sequence. In iMARGI, the enzyme restriction sequence is AGCT and it
         cuts between G and C, so the right offset is 2, then -C 2.
    -o : Output file name with directory. It&#39;s a .gz file (tabix indexed bgzip file).
    -h : Show usage help
</pre></div>
</div>
</div>
<div class="section" id="imargi-parse-sh">
<h2>imargi_parse.sh<a class="headerlink" href="#imargi-parse-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_parse.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: $PROGNAME [-r &lt;ref_name&gt;] [-c &lt;chromSize_file&gt;] [-R &lt;restrict_sites&gt;] [-b &lt;bam_file&gt;] [-o &lt;output_dir&gt;] 
                     [-Q &lt;min_mapq&gt;] [-G &lt;max_inter_align_gap&gt;] [-O &lt;offset_restriction_site&gt;] [-M &lt;max_ligation_size&gt;]
                     [-d &lt;drop&gt;] [-D &lt;intermediate_dir&gt;] [-t &lt;threads&gt;] 
    
    Dependency: pairtools pbgzip

    This script will use pairtools to parse the BAM alignments to interaction read pairs in .pairs format, and apply
    de-duplication and filtering.

    -r : Reference assembly name, such as &quot;hg38&quot;
    -c : Chromosome size file.
    -R : DNA restriction enzyme digestion sites bed file.
    -b : BAM file generated by &quot;bwa mem -SP5M&quot; mapping of iMARGI data.
    -o : Output directoy
    -Q : Min MAPQ value, default 1.
    -G : Max inter align gap for pairtools parsing. Default 20. It will allow R1 5&#39; end clipping.
    -O : Max mis-offset bases for filtering pairs based on R2 5&#39; end positions to restriction sites. Default 3.
    -M : Max size of ligation fragment for sequencing. It&#39;s used for filtering unligated DNA sequence. Default 1000.
    -d : Flag of dropping. Default is false, i.e., output all the intermediate results.
    -D : Directory for intermediate results. Works when -d false. Default is a sub-folder &quot;intermediate_results&quot; 
         in output directory.
    -t : Max CPU threads for parallelized processing, at least 4. (Default 8)
    -h : Show usage help
</pre></div>
</div>
</div>
<div class="section" id="imargi-stats-sh">
<h2>imargi_stats.sh<a class="headerlink" href="#imargi-stats-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_stats.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: <span class="nv">$PROGNAME</span> <span class="o">[</span>-D &lt;distance_type&gt;<span class="o">]</span> <span class="o">[</span>-d &lt;distance_threshold&gt;<span class="o">]</span> <span class="o">[</span>-i &lt;input_file&gt;<span class="o">]</span> <span class="o">[</span>-o &lt;output_file&gt;<span class="o">]</span>

    Dependency: gzip, awk
    This script can be used to filter out short-range intra-chromosomal interactions with a threshold genomic distance.

    -D : Distance type. The default genomic position in .pairs file is the <span class="m">5</span><span class="s1">&#39; end position, so the default distance</span>
<span class="s1">         type is the distance between 5&#39;</span> end of R1 and R2, which meand <span class="s1">&#39;-D 5end&#39;</span>. The alternative choices are <span class="s1">&#39;outer&#39;</span>
         and <span class="s1">&#39;inner&#39;</span>. <span class="s1">&#39;outer&#39;</span> means the outer-side distance between R1 and R2 <span class="nb">read</span> pairs. <span class="s1">&#39;inner&#39;</span> means the inner-side
         distance between R1 and R2 <span class="nb">read</span> pairs. Default is <span class="s1">&#39;5end&#39;</span>.
    -d : sets the genomic distance threshold of defining proximal and distal interactions <span class="o">(</span>intra-chromosomal<span class="o">)</span>. It will
         report the number of proximal and distal interactions. Besides, you can <span class="nb">set</span> a list of values separated with
         comma <span class="s1">&#39;,&#39;</span>, such as <span class="s1">&#39;-d 1000,2000,10000,20000,100000,1000000&#39;</span>, <span class="k">then</span> the report will include the statistics
         number with different thresholds <span class="o">(</span>space is not allowed<span class="o">)</span>.
    -i : Input .pairs.gz file.
    -o : Output stats text file.
    -h : Show usage <span class="nb">help</span>
</pre></div>
</div>
</div>
<div class="section" id="imargi-distfilter-sh">
<h2>imargi_distfilter.sh<a class="headerlink" href="#imargi-distfilter-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_distfilter.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: <span class="nv">$PROGNAME</span> <span class="o">[</span>-D &lt;distance_type&gt;<span class="o">]</span> <span class="o">[</span>-d &lt;distance_threshold&gt;<span class="o">]</span> <span class="o">[</span>-F &lt;deal_with_filter&gt;<span class="o">]</span> <span class="o">[</span>-i &lt;input_file&gt;<span class="o">]</span> 
                     <span class="o">[</span>-o &lt;output_file&gt;<span class="o">]</span>

    Dependency: gzip, awk
    This script can be used to filter out short-range intra-chromosomal interactions with a threshold genomic distance.

    -D : Distance type. The default genomic position in .pairs file is the <span class="m">5</span><span class="s1">&#39; end position, so the default distance</span>
<span class="s1">         type is the distance between 5&#39;</span> end of R1 and R2, which meand <span class="s1">&#39;-D 5end&#39;</span>. The alternative choices are <span class="s1">&#39;outer&#39;</span>
         and <span class="s1">&#39;inner&#39;</span>. <span class="s1">&#39;outer&#39;</span> means the outer-side distance between R1 and R2 <span class="nb">read</span> pairs. <span class="s1">&#39;inner&#39;</span> means the inner-side
         distance between R1 and R2 <span class="nb">read</span> pairs. Default is <span class="s1">&#39;5end&#39;</span>.
    -d : The distance threshold <span class="k">for</span> filtering. Default is <span class="m">200000</span> <span class="o">(</span>distance &lt;<span class="m">200000</span> will be filtered out<span class="o">)</span>.
    -F : How to deal with the interactions need to be filtered out? <span class="s1">&#39;-F&#39;</span> accepts <span class="s1">&#39;drop&#39;</span> and <span class="s1">&#39;output&#39;</span>. <span class="s1">&#39;drop&#39;</span> means
         drop those interactions. <span class="s1">&#39;output&#39;</span> means output an new file of all the filtered out interactions with prefix
         <span class="s1">&#39;filterOut_&#39;</span>. Default is <span class="s1">&#39;output&#39;</span>.
    -i : Input .pairs.gz file.
    -o : Output .pairs.gz file.
    -h : Show usage <span class="nb">help</span>
</pre></div>
</div>
</div>
<div class="section" id="imargi-convert-sh">
<h2>imargi_convert.sh<a class="headerlink" href="#imargi-convert-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_convert.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: $PROGNAME [-f &lt;file_format&gt;] [-k &lt;keep_cols&gt;] 
                     [-b &lt;bin_size&gt;] [-r &lt;resolution&gt;] [-T &lt;transpose&gt;] 
                     [-i &lt;input_file&gt;] [-o &lt;output_file&gt;] 

    Dependency: gzip, awk, cool
    This script can convert .pairs format to BEDPE, .cool, and GIVE interaction format.
    -f : The target format, only accept &#39;cool&#39;, &#39;bedpe&#39; and &#39;give&#39;. For &#39;cool&#39;, it will generate a &quot;.cool&quot; file
         with defined resolution of -b option and a -r defined multi-resolution &quot;.mcool&quot; file based on the &quot;.cool&quot; file.
         For &#39;bedpe&#39;, the output will be pbgzip compressed file. So keep in mind to name the output_file &#39;-o&#39; with
         &#39;.gz&#39; extesion. For &#39;give&#39;, the output is a normal text file.
    -k : (Only for BEDPE) Keep extra information column in BEDPE. Columns ids in .pairs file you want to keep.
         For example, &#39;cigar1,cigar2&#39;. Default value is &quot;&quot;, i.e., drop all extra cols.
    -b : (Only for cool/mcool) bin size for cool format. Default is 1000.
    -r : (Only for cool/mcool) resolution for cool/mcool format. Integers separated by comma. The values of resolution
         must be integer multiples of the bin size defined by -b option.
         Default is 1000,2000,5000,10000,25000,50000,100000,250000,500000,1000000,2500000,5000000,10000000
    -T : (Only for cool/mcool) mcool file can be visualized by HiGlass. Currently the heatmap orientation cannot be
         set in HiGlass control panel. So if you want to transpose the interaction map in HiGlass, you need to generate
         a transposed mcool file. The default value is &#39;fasle&#39;, i.e., no transpose, the RNA-DNA interactions will be
         mapped to a X-Y system as a DNA x RNA contact matrix. If set &#39;-T true&#39;, then the generated cool/mcool file is 
         transposed, which is RNA x DNA contact matrix.
    -i : Input file.
    -o : Output file.
         BEDPE output is gzip compressed file, so it&#39;s better to have a .gz file extension.
         cool output are two files, .cool and .mcool. The -o option assigns the name of .cool file, it must use .cool as
         extension. The .mcool file will be generated based on the .cool file with .mcool extension.
    -h : Show usage help
</pre></div>
</div>
</div>
<div class="section" id="imargi-annotate-sh">
<h2>imargi_annotate.sh<a class="headerlink" href="#imargi-annotate-sh" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/Zhong-Lab-UCSD/iMARGI-Docker/blob/master/src/imargi_annotate.sh"><em>Source Code</em></a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>    Usage: imargi_annotate.sh [-A &lt;annotation_format&gt;] [-a &lt;annotation_file&gt;] 
                [-l &lt;annotation_level&gt;] [-f &lt;feature_attribute&gt;]
                [-C &lt;end_for_annotate&gt;] [-c &lt;add_col_names&gt;] [-s &lt;strand_specific&gt;]
                [-m &lt;min_overlap&gt;] [-G &lt;cigar&gt;]
                [-t &lt;threads&gt;] [-i &lt;input_file&gt;] [-o &lt;output_file&gt;]

    Dependency: gzip, pairtools, lz4 pbgzip

    This script can annotate both RNA and DNA ends with gene annotations in GTF/GFF format or any other genomic
    features in a simple BED file (each line is a named genomic feature). Multiple overlapped annotation features are
    separated by &#39;,&#39;.

    -A : Only accept &#39;-A gtf&#39; for annotating gene with GTF file or &#39;-A bed&#39; for annotating any other genomic features
         in a simple BED format file, whose 4th column is the feature name. Default is &#39;-A gtf&#39;.
    -a : The annotation information file.
    -l : Only work with &#39;-A gtf&#39;. Gene annotation level, &#39;exon&#39; or &#39;gene&#39;. Default is &#39;-l gene&#39;.
    -f : Only work with &#39;-A gtf&#39;. There are many gene feature attributes can be added, such as gene_id, gene_name,
         or gene_type. Use &#39;,&#39; to seperate the features. In the output results, these features are seperated with
         &#39;|&#39;. Default is &#39;-f gene_id,gene_name,gene_type&#39;. Blank space is not allowed!
    -C : Determine Which end to be annotated. It accepts &#39;-C RNA&#39;, &#39;-C DNA&#39; or &#39;-C both&#39;. Default is &#39;-C both&#39;,
         then both RNA and DNA ends will be annotated. When use &#39;-C RNA&#39; or &#39;-C DNA&#39;, then &#39;-c&#39;, &#39;-s&#39;, &#39;-m&#39; and &#39;-G&#39;
         only need one value, if you provided two, then only the first one will be used. If use &#39;-C both&#39;, then you must
         provide two values for &#39;-c&#39;, &#39;-s&#39;, &#39;-m&#39; and &#39;-G&#39;.
    -c : The names of annotation column in the output file. If &#39;-C both&#39;, then you need set two column names separated
         with comma &#39;,&#39;, i.e., &#39;-c gene1,gene2&#39;. Blank space is not allowed!
    -s : Strand specific annotation option, &#39;s&#39;, &#39;r&#39; and &#39;n&#39;. &#39;s&#39; means strand specific, &#39;r&#39; means
         reverse-strand-specific, and &#39;n&#39; means none strand specific, ignoring it. As RNA end of iMARGI is
         reverse-strand-specific, and DNA end is none strand specific, so the default is &#39;-s rn&#39;.
    -m : Minimum bases of overlapping for annotation. &#39;-m 1&#39; means at least one base overlapping. Default &#39;-m 1,1&#39;.
         &#39;-m 0&#39; means it must be totally inside of the genomic feature.
    -G : Use CIGAR columsn information . Default value is use standard CIGAR info columns, i.e., &#39;-G cigar1,cigar2&#39;.
         Blank space is not allowed! If you just want to use the 5&#39; end genomic coordinate reported in .pairs file for
         annotation, set &#39;-G false,false&#39; and the &#39;-m&#39; will be automatically set as &#39;0,0&#39;.
    -t : Max CPU threads for parallelized processing, at least 4. (Default 8)
    -i : Input file.
    -o : Output file.
    -h : Show usage help
</pre></div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="faq_win_mac.html" class="btn btn-neutral" title="Guides for Issues on Windows and macOS System" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Zhong Lab, UCSD. iMARGI-Docker code licensed under the BSD 2 License. Documentation licensed under CC BY 3.0

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    
    
      <script type="text/javascript">
          var DOCUMENTATION_OPTIONS = {
              URL_ROOT:'./',
              VERSION:'1.1',
              LANGUAGE:'None',
              COLLAPSE_INDEX:false,
              FILE_SUFFIX:'.html',
              HAS_SOURCE:  true,
              SOURCELINK_SUFFIX: '.txt'
          };
      </script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>